scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/factories/spi/JobValidatorService.java

```diff
@@ -45,18 +45,20 @@
      * Example of use: xml schema validation.
      *
      * @param jobFile xml job file to validate
+     * @return if the validator eventually made some modifications to the job, return a new version
      * @throws JobValidationException if the job is not valid
      */
-    void validateJob(File jobFile) throws JobValidationException;
+    File validateJob(File jobFile) throws JobValidationException;
```

Comment:
Here is an example of validation methods which return objects https://commons.apache.org/proper/commons-lang/javadocs/api-3.1/org/apache/commons/lang3/Validate.html